[docs] update access log filters guide for usage clarity (#9998)
solo-io/gloo: docs/content/guides/security/access_logging/_index.md

               comparison:
                 op: GE
                 value: 
                   defaultValue: 400
                   # see note below about runtimeKey
                   runtimeKey: "my_status_code_filter"
     proxyNames:
     - gateway-proxy
     ssl: false
     useProxyProto: false
   ```

{{% notice note %}}

> Note that `runtimeKey` is only effective if Envoy is configured appropriately with a runtime value of matching key. This can be achieved with the admin interface of the gateway proxy or via the Helm value [gatewayProxies.NAME.customStaticLayer]({{< versioned_link_path fromRoot="/reference/helm_chart_values/" >}}).

Reviewer: ```suggestion
Note that the `runtimeKey` is enforced only if it matches a key that is defined in Envoy's runtime configuration layer. Gloo Gateway does not include a key by default. To specify a key-value pair, use the [gatewayProxies.NAME.customStaticLayer]({{< versioned_link_path fromRoot="/reference/helm_chart_values/" >}}) Helm value or set the key at runtime by using the gateway proxy admin interface.
```

               comparison:
                 op: GE
                 value: 

>                    default: 400

Reviewer 2: i may be looking at the wrong proto but should it be `defaultValue`? https://github.com/solo-io/gloo/blob/29fdbaa99fb16c31a3b976c4ac4598dd293b9ac3/projects/gloo/api/external/envoy/config/core/v3/base.proto#L237

Author: https://github.com/solo-io/gloo/blob/29fdbaa99fb16c31a3b976c4ac4598dd293b9ac3/install/helm/gloo/crds/gateway.solo.io_v1_ListenerOption.yaml#L155

oof good catch thanks

                   # Note that `runtimeKey` is only effective if Envoy is configured 

>                    # appropriately with a runtime value of matching key. This can be achieved

Reviewer: could we clarify what matching key is? Is that a field?

Reviewer: if it is an HTTP response code, maybe say `..is configured to match on a supported HTTP response code.` (and link here https://http.dev/status)

Author: the runtime key configured in envoy has to be what is specified here, but we don't set that. So it's only if a user has it configured that they will know an appropriate value for this field

>                    # with the admin interface of the gateway proxy or via the helm value 

Reviewer: ```suggestion
                   # with the admin interface of the gateway proxy or via the Helm value 
```

>                    # [gatewayProxies.NAME.customStaticLayer]({{< versioned_link_path fromRoot="/reference/helm_chart_values/" >}})

Reviewer: ```suggestion
                   # [gatewayProxies.NAME.customStaticLayer]({{< versioned_link_path fromRoot="/reference/helm_chart_values/" >}}).
```
